platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,373 @@
+
+The **[CI pipeline](#ci-pipeline)** runs this build script (among other things)
+whenever a commit is pushed to a branch in GitHub. On a branch, this build must
+be successful **before the branch can be merged.**
+
+The output of this build are **[deployed](#deployment) to and hosted from an AWS
+S3 bucket**. Applications are deployed on a weekdaily schedule. New content is
+deployed by a manual trigger from the CMS and by the weekdaily application
+deployment.
+
+The entangling of the content and application builds have [led to
+frustrations](#pain-points) for engineers, slower iteration cycles due to failed
+builds, and failed production deployments.
+
+#### Build script
+[Metalsmith](https://github.com/segmentio/metalsmith) is a static content build
+pipeline orchestrator. `vets-website` uses it to manage the steps in the content
+build. The highlights of what this script does include:
+- Fetching static content from Drupal
+- Building static HTML pages from the content
+- Triggering Webpack to build JS and CSS bundles
+- Creating the HTML pages for the React Single-Page Applications (SPAs)
+- Checking all HTML pages for broken links
+
+The script can be found [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/index.js).
+
+#### CI pipeline
+The CI pipeline for `vets-website` triggers the above build and runs a number of
+tests. Pertinent to this document are:
+- Accessibility tests on static content pages
+  - Using the [`npm run nightwatch:docker --
+    --env=accessibility`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/Jenkinsfile#L82) command
+  - This iterates through all the pages on the sitemap, which is created by the
+    content build script
+- Broken link checks on static content pages
+  - The actual link checking happens during the build script, but it isn't
+    reported anywhere until [after the build script has been run](https://github.com/department-of-veterans-affairs/vets-website/blob/master/jenkins/common.groovy#L200)
+    
+#### Deployment
+There are currently two kinds of deployments:
+1. [Partial deploy](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/workflow/deploy/#partial-deploy--static-page-changes-only)
+    - For static pages only
+    - Uses the latest `vets-website` release and static assets
+      - Does not trigger Webpack to build the JS and CSS bundles
+      - Instead,
+        [fetches](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/configure-assets.js#L22-L23)
+        the assets [from S3](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/site/stages/build/plugins/download-assets.js#L111)
+        - Process is triggered [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/Jenkinsfile.content#L32)
```

Comment:
tiny nit: each of these links is going to get out of date fast since you're linking to `master` with line numbers